### PR TITLE
Reflect the fact that VMsnapshots are already supported as clone source

### PR DESCRIPTION
**Pull request by iholder101**
**What this PR does / why we need it**:
Reflect the fact that VMsnapshots are already supported as clone source.

**Release note**:

```release-note
Reflect the fact that VMsnapshots are already supported as clone source
```

### CODE REVIEW

File: docs/storage/clone_api.md [hunk 1]
```diff
@@ -155,28 +157,18 @@ After the clone is finished, the target can be inspected:
...
 * Create a VM
 
-* Prepare a "golden VM" environment
-
-  * This can mean different things in different contexts. For example, write files, install applications, apply configurations,
-    or anything else.
+* Prepare a "golden VM" environment. This can mean different things in different contexts.
+For example, write files, install applications, apply configurations, or anything else.
 
 * Snapshot the VM
```
Comment (ShellyKa13) [inline docs/storage/clone_api.md:174]: why do you have to delete the vm?

File: docs/storage/clone_api.md [hunk 2]
```diff
@@ -155,28 +157,18 @@ After the clone is finished, the target can be inspected:
...
 **Create a golden image**
 
 * Create a VM
 
-* Prepare a "golden VM" environment
-
-  * This can mean different things in different contexts. For example, write files, install applications, apply configurations,
-    or anything else.
+* Prepare a "golden VM" environment. This can mean different things in different contexts.
+For example, write files, install applications, apply configurations, or anything else.
```
Comment (ShellyKa13) [inline docs/storage/clone_api.md:171]: I dont quite understand this examples so maybe worth to add a few words or explanations

File: docs/storage/clone_api.md [hunk 3]
```diff
@@ -155,28 +157,18 @@ After the clone is finished, the target can be inspected:
 kubectl get vm vm-clone-target -o yaml
 ```
 
-## Future roadmap
-
-The clone API is in an early alpha version and may change dramatically. There are many improvements and features
-that are expected to be added, the most significant goals are:
-
-* Add more supported source types like `VirtualMachineInstace` in the future.
-* Add a cross-namespace clone support. This needs to be supported for snapshots / restores first.
```
Comment (ShellyKa13) [inline docs/storage/clone_api.md:164]: are this 2 goals not expected to happen anymore?

File: docs/storage/clone_api.md [hunk 4]
```diff
@@ -1,7 +1,10 @@
 # Clone API
 
-The `clone.kubevirt.io` API Group defines resources for cloning KubeVirt objects. Currently, the only supported cloning
-type is `VirtualMachine`, but more types are planned to be supported in the future (see future roadmap below).
+The `clone.kubevirt.io` API Group defines resources for cloning KubeVirt objects.
+
+This API is useful both for cloning Virtual Machines,
+but also for creating "golden snapshots" that would be used as a template for creating new VMs
```
Comment (ShellyKa13) [inline docs/storage/clone_api.md:6]: In my opinion not sure 'but' is the best word here..
'but' is more fitting if you use 'not only for....' 'but also'
here you put both for ...., but also
seems a bit odd.

Comment (ShellyKa13) [reply docs/storage/clone_api.md:6]: Also I think you should distinguish between the possible sources of the clone and the use cases of it, and probably put those in different paragraphs :)

**Review by ShellyKa13 — COMMENTED** [review]
Hi thanks for the update, some comments inline

File: docs/storage/clone_api.md (hunk 1 above)
Comment (iholder101) [reply docs/storage/clone_api.md:175]: I don't :)
Added that it's optional to delete it

File: docs/storage/clone_api.md (hunk 2 above)
Comment (iholder101) [reply docs/storage/clone_api.md]: I've added a more specific example. WDYT?

File: docs/storage/clone_api.md (hunk 3 above)
Comment (iholder101) [reply docs/storage/clone_api.md:164]: > Add more supported source types like `VirtualMachineInstace` in the future.

Not really relevant anymore now that the snapshot source is supported.

> Add a cross-namespace clone support. This needs to be supported for snapshots / restores first.

As written, it has to be supported by snapshot/restore first, and I'm not aware of a concrete plan to support it. Perhaps in the further future.

**Comment by iholder101** [comment]
Thanks @ShellyKa13 for your review!

@jean-edouard do you mind having a look as well?

**Review by jean-edouard — COMMENTED** [review]
/approve

**Pull request merged**